python/tank/authentication/shotgun_shared/saml2_sso.py

```diff
@@ -0,0 +1,576 @@
+
+            # For debugging purposes
+            # @TODO: Find a better way than to use this EV
+            if "SAML2_SHOTGUN_USING_VM" in os.environ:
+                interval = 5000
+        log.debug("==- start_sso_renewal: interval: %s" % interval)
+
+        self._sso_countdown_timer.setInterval(interval)
+        self._sso_countdown_timer.start()
+        self._session_renewal_active = True
+
+    def on_http_response_finished(self, reply):
+        """on_http_response_finished."""
+        # log.debug("==- on_http_response_finished")
+
+        error = reply.error()
+        url = reply.url().toString()
+        session = AuthenticationSessionData() if self._session is None else self._session
+
+        if (
+            error is not QtNetwork.QNetworkReply.NetworkError.NoError and
+            error is not QtNetwork.QNetworkReply.NetworkError.OperationCanceledError
+        ):
+            if error is QtNetwork.QNetworkReply.NetworkError.HostNotFoundError:
+                session.error = HTTP_CANT_CONNECT_TO_SHOTGUN
+            elif error is QtNetwork.QNetworkReply.NetworkError.ContentNotFoundError:
+                if url.startswith(session.host + URL_SAML_RENEW_PATH):
+                    # This is likely because the subdomain is not valid.
+                    # e.g. https://foobar.shotgunstudio.com
+                    # Here the domain (shotgunstudio.com) is valid, but not
+                    # foobar.
+                    session.error = HTTP_CANT_CONNECT_TO_SHOTGUN
+                else:
+                    # We silently ignore content not found otherwise.
+                    pass
+            elif error is QtNetwork.QNetworkReply.NetworkError.UnknownContentError:
+                # This means that the site does not support SSO or that
+                # it is not enabled.
+                session.error = HTTP_AUTHENTICATE_SSO_NOT_UPPORTED
+            elif error is QtNetwork.QNetworkReply.NetworkError.ContentOperationNotPermittedError:
+                # This means that the SSO login worked, but that the user does
+                # have access to the site.
+                session.error = HTTP_CANT_AUTHENTICATE_SSO_NO_ACCESS
+            else:
+                session.error = reply.attribute(QtNetwork.QNetworkRequest.HttpReasonPhraseAttribute)
+        elif url.startswith(session.host + URL_LOGIN_PATH):
+            # If we are being redirected to the login page, then SSO is not
+            # enabled on that site.
+            session.error = HTTP_AUTHENTICATE_SSO_NOT_UPPORTED
+
+        if session.error:
+            # If there are any errors, we exit by force-closing the dialog.
+            log.error("==- on_http_response_finished: %s - %s" % (url, error))
+            self._dialog.reject()
+
+    def is_handling_event(self):
+        """
+        Called to know if an event is currently being handled.
+        """
+        # log.debug("==- is_handling_event")
```

Comment:
I will refactor my code to bundle the event-based methods in a RV-specific wrapper.

For the moment, I will leave this in to minimize the delta with the code in RV.